rfcs/0049-flakes.md

```diff
@@ -0,0 +1,703 @@
+---
+feature: flakes
+start-date: 2019-07-09
+author: Eelco Dolstra
+co-authors: TBD
+shepherd-team: Domen Kožar, Alyssa Ross, Shea Levy, John Ericson
+shepherd-leader: Domen Kožar
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+This RFC proposes a mechanism to package Nix expressions into
+composable entities called "flakes". Flakes allow hermetic,
+reproducible evaluation of multi-repository Nix projects; impose a
+discoverable, standard structure on Nix projects; and replace previous
+mechanisms such as Nix channels and the Nix search path.
+
+# TLDR
+
+* A prototype implementation is available in Nixpkgs: `nix run
+  nixpkgs.nixFlakes`.
+
+* Flakes replace channels. For example,
```

Comment:
Are they supposed to also fit some of the overlay use cases?

Reply:
Not really. They can be a delivery mechanism for overlays, though.